docs/sources/configure-client/grafana-agent/java/_index.md

```diff
@@ -78,6 +78,27 @@ see [profiler-options](https://github.com/async-profiler/async-profiler?tab=read
 You must run the collector, either Grafana Alloy (recommended) or Agent (legacy), as root and inside host `pid` namespace for the `pyroscope.java`
 and `discover.process` components to work.
 
+### Additional Configuration for Linux Capabilities
+If your Kubernetes environment has Linux capabilities enabled, configure the following in the Helm values to ensure `pyroscope.java` functions properly:
+
+```yaml
+alloy:
+  securityContext:
+    runAsUser: 0
+    runAsNonRoot: false
+    capabilities:
+      add:
+        - PERFMON
+        - SYS_PTRACE
+        - SYS_RESOURCE
+        - SYS_ADMIN
+```
```

Comment:
Technical question: If we are setting `runAsUser: 0` and `runAsNonRoot: false`, does the process not already have complete capabilities as the process is running as the root user?

Reply:
For context, this comes from https://github.com/grafana/alloy/issues/1616.

In Kubernetes, `runAsUser: 0` and `runAsNonRoot: false` run the container as root. However, root privileges don't automatically grant all capabilities in containerized environments. The capabilities setting adds specific capabilities, but its exact behavior depends on the Container Runtime Interface (CRI) used.

AFAIK, in Docker, "The operator can whitelist specific capabilities, and any capabilities not whitelisted are dropped from the container." However, other CRIs may behave differently... 🤔 

Maybe adding a comment such as:
```yaml
# Note: Capability behavior depends on CRI settings. In Docker, non-whitelisted capabilities are dropped by default.
```

Reply:
Thanks for explaining :)

Reply:
If you choose to add the note, consider changing whitelisted to allowlisted instead.